### DISCUSSION TITLE
"> Task :checkRelease FAILED whatsnew and changelog are not in sync"

### DISCUSSION THREAD

**Discussion by fatboychummy** [cc-tweaked/CC-Tweaked#810] [Q&A] [answered]
Attempting to build `gradlew.bat build` (or even just run the tests `gradlew.bat check`) on my windows installation, but getting this error. I've looked into `build.gradle` and noticed that it's trying to check if the `whatsnew.txt` file in rom is contained within the `changelog.txt` file (prepending `# ` and removing the changelog mention at the end to the `whatsnew.txt` file to make it equal). It all looks good (and I'll throw those files here), but for some reason it is failing.

The mod version in `gradle.properties` is 1.96.0.

[changelog.txt](https://github.com/SquidDev-CC/CC-Tweaked/files/6603183/changelog.txt)
[whatsnew.txt](https://github.com/SquidDev-CC/CC-Tweaked/files/6603184/whatsnew.txt)

I've not got a lot of experience using gradle, is there something I'm missing here?

**Comment by SquidDev** [comment, marked as answer]
This is a line ending issue caused by Git checking out files with crlf, rather than the correct lf. There's a config option you can change to fix this, but I'm afraid I cannot remember what it is - IIRC there's several relating to line feeds, and they all do subtly different things.

That said, we should probably relax the changelog check so it can handle different line endings.

**Reply by fatboychummy** [reply]
This worked.

`git config --global core.autocrlf input`

Disables conversion to CRLF on files that weren't already CRLF to begin with, not sure why it's on by default, but Windows will be Windows...

**Comment by SquidDev** [comment]
As an aside `./gradlew test` will run the tests without checking the changelog - might be more useful in general.

**Reply by fatboychummy** [reply]
Ah, that will be useful. I'll check it out in a few hours here.

Thank you for the information.
